hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationRunnerImpl.java

```diff
@@ -131,7 +133,9 @@ public void run(Runnable task) {
     }
 
     private boolean publishCurrentTask() {
-        return (getPartitionId() != AD_HOC_PARTITION_ID && currentTask == null);
+        boolean isClientRunnable = currentTask instanceof MessageTask
```

Comment:
@gurbuzali 
All client runnables has to be MessageTask in new protocol. And all old client requests are handled by single runnable class ClientPacketProcessor(which we will get rid of in 3.7 SNAPSHOT) . So in theory, there should not be a problem because of an addition of new client runnable.